lhotse/supervision.py

```diff
@@ -251,12 +254,49 @@ def from_segments(segments: Iterable[SupervisionSegment]) -> 'SupervisionSet':
     def from_dicts(data: Iterable[Dict]) -> 'SupervisionSet':
         return SupervisionSet.from_segments(SupervisionSegment.from_dict(s) for s in data)
     
-    def add_alignments_from_ctm(ctm) -> 'SupervisionSet':
+    def add_alignments_from_ctm(self, ctm_file: Pathlike, type: str = 'word') -> 'SupervisionSet':
         """
+        Add alignments from CTM file to the supervision set.
         
-        """
-        pass
-
+        :param ctm: Path to CTM file.
+        :param type: Alignment type (optional, default = `word`).
+        :return: A new SupervisionSet with AlignmentItem objects added to the segments.
+        """
+        ctm_words = []
+        with open(ctm_file, 'r') as f:
+            for line in f:
+                reco_id, channel, start, duration, symbol = line.strip().split()
+                ctm_words.append((reco_id, channel, float(start), float(duration), symbol))
+        ctm_words = sorted(ctm_words, key=lambda x:x[0])
```

Comment:
is python's `sorted` guaranteed to be stable? otherwise you might want to change the key to be a tuple of `(reco_id, start)`

Reply:
Yes, it is guaranteed (https://docs.python.org/3/library/functions.html#sorted). Also, we don't particularly need the alignments to be sorted by time I suppose? This sorting by `reco_id` is just to enable `groupby`.

Reply:
We probably don't, but I think it's a reasonable expectation that they are sorted (I know I'd be surprised if they're not). If it's guaranteed then it's cool :)